pkgs/development/python-modules/xc-fasm/default.nix

```diff
@@ -0,0 +1,41 @@
+{ stdenv
+, fetchFromGitHub
+, buildPythonPackage
+, simplejson
+, intervaltree
+, python-prjxray
+, symbiflow-fasm
+, textx
+}:
+
+buildPythonPackage rec {
+  pname = "xc-fasm";
+  version = "0.0";
+
+  src = fetchFromGitHub {
+    owner = "SymbiFlow";
+    repo = "xc-fasm";
+    rev = "0ddd951602d47d5b95f2072f8aa751af5e81e577";
+    sha256 = "15bzw92sx99s0zldr48na4yhrnp7b90nxsd8ya6ag1pvvijp2al4";
+  };
+
+  propagatedBuildInputs = [
+    simplejson
+    intervaltree
+    python-prjxray
+    symbiflow-fasm
+    textx
+  ];
+
+  # Pip will check for and then install missing dependecies.
+  # Because some of them are installed from git, it will try
+  # to download them even if they're present in buildInputs.
+  pipInstallFlags = [ "--no-deps" ];
```

Comment:
do you mind adding a `pythonImportsCheck` as a sanity check?

If tests are not available, then please use `pythonImportsCheck` to import the most important modules. This isn't as good as unit tests, but will usually give a good indication of run-time errors. Please see [pythonImportsCheck documentation](https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonimportscheck) for more information.